[Yuichiro Fukushima] iP

yuifuku1118 opened the pull request.

yuifuku1118 committed:
level1

yuifuku1118 committed:
level2

yuifuku1118 committed:
level3

yuifuku1118 committed:
level4

yuifuku1118 committed:
level5

yuifuku1118 committed:
level6

yuifuku1118 committed:
A-TextUiTesting

yuifuku1118 committed:
A-Enums

yuifuku1118 committed:
Level-7

yuifuku1118 committed:
level 7 on branch

yuifuku1118 committed:
level 8 on branch

yuifuku1118 committed:
More oop

yuifuku1118 committed:
Made a duke package

yuifuku1118 committed:
unit-test

yuifuku1118 committed:
unit-test

yuifuku1118 committed:
creating jar file

yuifuku1118 committed:
added javadoc

yuifuku1118 committed:
added find

yuifuku1118 committed:
mergin

dt-td commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,70 @@
+            return Action.UNKNOWN;
+        }
+    }
+
+    public Action getAction(){
+        return this.action;
+    }
+
+    public int getCommandSize(){
+        String[] splited = this.command.split(" ", 2);
+        return splited.length;
+    }
+
+    public String getActionList(){
+        String[] splited = this.command.split(" ", 2);
+        return splited[1];
+    }
+
```
I like how you named your methods with verbs. At one glance I can understand what your method does easily. 👍

dt-td commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+        catch(InputNotValidError e){
+            String error = 
+            "   ____________________________________________________________\n" +
+            "   " + e.getMessage() + '\n' +
+            "   ____________________________________________________________";
+            System.out.println(error);
+            return;
+        }
+        if (validCommand){
+            Action actionName = actionList.getAction();
+
+            switch(actionName){
+                case LIST:
+                    this.printList();
+                    break;
+                case DONE:
+                    int num = Integer.parseInt(actionList.getActionList());
+                    this.done(num);
+                    break;                
+                case TODO:
+                    this.addToDo(actionList.getActionList(), done, fromData);
+                    break;
+                case DEADLINE:
+                    this.addDeadline(actionList.getActionList(), done, fromData);
+                    break;
+                case EVENT:
+                    this.addEvent(actionList.getActionList(), done, fromData);
+                    break;
+                case DELETE:
+                    int num2 = Integer.parseInt(actionList.getActionList());
+                    this.delete(num2);
+                    break;
+                case FIND:
+                    this.find(actionList.getActionList());
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
```
The indentations inside the switch block is incorrect. You might want to look at the coding standard document here: https://se-education.org/guides/conventions/java/intermediate.html

dt-td commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+            }
+        }
+    }
+    
+    public void addToDo(String action, boolean done, boolean fromdata){
+        ToDo newTask = new ToDo(action, done, "T");
+        this.addTask(newTask, fromdata);
+    }
+
```
Perhaps you might want to consider following camelCase standard in the naming of your method arguments too.

dt-td commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+        }
+    }
+
+    public void LoadTask(ArrayList<String[]> data) throws InputNotValidError{
+        for (String[] each : data){
+            String[] actions = new String[2];
+            String actionName = each[0];
+            Boolean done = true;
+            if(each[1].equals(" 0 ")){
+                done = false;
+            }
+            if (actionName.equals("T ")){
+                actions[0] = "todo";
+                actions[1] = each[2].substring(1, each[2].length());
+            }else if(actionName.equals("E ")){
+                actions[0] = "event";
+                actions[1] = each[2].substring(1) + "/at"  + each[3].substring(1, each[3].length());
+            }else if(actionName.equals("D ")){
+                actions[0] = "deadline";
+                actions[1] = each[2].substring(1) + "/by" + each[3].substring(1, each[3].length());                
+            }
+            Parser hitoryData = new Parser(actions[0] + " " + actions[1]);
+            this.actionHalder(hitoryData, done, true);
+        }
```
Perhaps you might want to consider following camelCase standard in the naming of your methods too.

dt-td commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+    return res;
+    }
+
+    /**
+     * Method to save data to txt file,
+     * @param tasks String array representation of tasks
+     */     
+    public void saveData(ArrayList<Task> tasks) throws IOException{
+        File fout = new File(this.filepath);
+        FileOutputStream fos = new FileOutputStream(fout);
+        
+        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos));
+        
+        for(Task eachtask : tasks){
+            String actionname = eachtask.getActionName();
+            boolean compleated = eachtask.getCompleted();
+            String type = eachtask.getType();
```
I like how you distinguished clearly between single-valued and multi-valued variables. 👍

dt-td reviewed (commented):
LGTM! Just some nits to fix. 👍

kaixin-hc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,65 @@
-        System.out.println("Hello from\n" + logo);
+
+    private Storage storage;
+    private TaskList tasklist;
```
Perhaps you should change the name to taskList to comply with the coding standards!

kaixin-hc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,65 @@
+        while(true){
+            System.out.println("");
+            String action = input.nextLine();
+            Parser commandpraser = new Parser(action);
```
I think you might have made a typo here? Did you mean `commandParser`, not `commandpraser` ?

kaixin-hc commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,90 @@
+  */
+public class Storage {
+
+    String filepath;
```
Perhaps you could consider using camelCase for all variable names to comply with the coding standard!

kaixin-hc commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+    /**
+     * Defult constructor,
+     */
+    public Deadline(String actionName, boolean compleated, String date, String type) {
```
I think you might have typo-ed completed as compleated here. Also, the coding standard suggests using prefixes like is, has, and was for boolean variable/method names - do you want to change this to `isCompleted`?

kaixin-hc commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+        Task task = this.list.get(num  - 1);
+        task.doneTask();
+        this.list.set(num  - 1, task);
+        String item = "\n" +
```
I like how all of Duke's responses are bracketed by lines, it looks very neat!

kaixin-hc commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,193 @@
+
+public class TaskList {
+    ArrayList<Task> list;
+    int curSize;
```
Since calling `size()` on arrayList runs in constant time, is there any reason why you decided to store this in a separate variable?

yuifuku1118 committed:
adding GUI

yuifuku1118 committed:
fixing GUT

yuifuku1118 committed:
fixing state

yuifuku1118 committed:
fixing style

yuifuku1118 committed:
Hu

yuifuku1118 committed:
adding prioority feture

yuifuku1118 committed:
adding feetures

yuifuku1118 committed:
adding tag that I forgot

yuifuku1118 committed:
Adding help

Current Situation: There is no help function \n Why Change : SO that first time user is able to see what's avilable \n What is done : I have implemented new action help

yuifuku1118 committed:
fixed readme and UG

yuifuku1118 committed:
final release

yuifuku1118 committed:
final version

yuifuku1118 committed:
fixing Ui.png
